projects/js-packages/connection/hooks/use-product-checkout-workflow/index.jsx

```diff
@@ -46,15 +52,55 @@ export default function useProductCheckoutWorkflow( {
 		from,
 	} );
 
-	// Build the checkout URL.
-	const checkoutProductUrl = getProductCheckoutUrl(
-		productSlug,
+	const checkoutUrl = useMemo( () => {
+		const origin = getCalypsoOrigin();
+		const shouldConnectAfterCheckout =
+			( ! isRegistered || ! isUserConnected ) && connectAfterCheckout;
+
+		const checkoutPath = shouldConnectAfterCheckout
+			? 'checkout/jetpack/'
+			: `checkout/${ siteSuffix }/`;
+
+		const productCheckoutUrl = new URL( `${ origin }${ checkoutPath }${ productSlug }` );
+
+		if ( shouldConnectAfterCheckout ) {
+			productCheckoutUrl.searchParams.set( 'connect_after_checkout', true );
+			productCheckoutUrl.searchParams.set( 'admin_url', adminUrl );
+			/**
+			 * `from_site_slug` is the Jetpack site slug (siteSuffix) passed 'from the site' via url
+			 * query arg (into	Calypso), for use cases when there is not a site in context (such
```

Comment:
This is probably a minor typo with a tab instead of space?

Reply:
Yes, fixed. 👍🙂